### PR TITLE
Fixed EateryPageVC bug

### PR DESCRIPTION
**Pull request by antoinztte**
## Overview

- Fixed bug so that the first page you see when you click on an eatery in the carousel now correctly corresponds to the chosen eatery
- Fixed recurring issue of "Implicit use of 'self' in closure" that occurs whenever app is running

## Changes Made

- Create new var 'index' in EateryModel
- Manually assigned index to each eatery in DummyData array
- Initialized private var selectedIndex in EateryPageVC that is passed through setVC method 
- Added new parameter to EateryPageVC so that eatery.index is passed through as selectedIndex
- Added self in front of UpdateAllEateriesFromNetworking to get rid of warning

## Test Coverage

- Tested on iPhone 14 (all sizes).

## Next Steps 

- Figure out way to retrieve index with manually assigning it (so that the code is applicable beyond DummyData testing)

## Related PRs or Issues 

- PR #9

### CODE REVIEW

File: Eatery Blue/Eatery Blue/UI/HomeScreen/HomeModelController.swift [hunk 1]
```diff
@@ -264,7 +264,7 @@ class HomeModelController: HomeViewController {
         Task {
             await withTaskGroup(of: Void.self) { [self] group in
                 group.addTask {
-                    await updateAllEateriesFromNetworking()
+                    await self.updateAllEateriesFromNetworking()
                 }
```
Comment (vinnie4k) [inline Eatery Blue/Eatery Blue/UI/HomeScreen/HomeModelController.swift:268]: You are using a strong reference to `self` within the closure which can cause retain cycles. Change `[self] group in` to `[weak self] group in` to create a weak reference of this object when it gets captured. You will then need to change line 267 to `await self?.updateAllEateriesFromNetworking()`

**Review by vinnie4k — COMMENTED** [review]
Nice PR Antoinette!!

**Review by ReadeP — APPROVED** [review]
LGTM!

**Review by jjennifergu — APPROVED** [review]
Looks good, I agree with Vin's suggestion. Please merge this right after making that change. Great work!

**Pull request merged**